api/src/server.ts

```diff
@@ -124,27 +124,29 @@ passportConfig(passport);
 app.use(passport.initialize());
 app.use(passport.authenticate('session'));
 
-app.use(
-  '/ferry',
-  createProxyMiddleware({
-    target: 'http://graphql-engine:8080',
-    pathRewrite: { '^/ferry/': '/' },
-    ws: true,
-    xfwd: true,
-    onProxyReq(proxyReq, req) {
-      req.headers['X-Hasura-Role'] = req.isAuthenticated()
-        ? 'student'
-        : 'anonymous';
-      const hasuraRole = req.headers['X-Hasura-Role'] ?? 'anonymous'; // Default to 'anonymous'
-      proxyReq.setHeader('X-Hasura-Role', hasuraRole);
-    },
-  }),
-);
+// Ferry proxy
+const ferryProxy = createProxyMiddleware({
+  target: 'http://graphql-engine:8080',
+  pathRewrite: { '^/ferry/': '/' },
+  xfwd: true,
+});
+
+// Add the authentication header to the request
+// Proxy initial HTTP requests to Ferry
+app.use('/ferry', (req, res, next) => {
+  req.headers['X-Hasura-Role'] = req.isAuthenticated()
+    ? 'student'
+    : 'anonymous';
+  const hasuraRole = req.headers['X-Hasura-Role'] ?? 'anonymous'; // Default to 'anonymous'
+  req.headers['X-Hasura-Role'] = hasuraRole;
+  ferryProxy(req, res, next);
+});
```

Comment:
```suggestion
app.use('/ferry', (req, res, next) => {
  const hasuraRole = req.isAuthenticated()
    ? 'student'
    : 'anonymous';
  req.headers['X-Hasura-Role'] = hasuraRole;
}, ferryProxy);
```

Something like this but with formatting

Reply:
Oh damn I didn't even know you could do this.